saleor/csv/tasks.py

```diff
@@ -59,3 +68,30 @@ def export_gift_cards_task(
 ):
     export_file = ExportFile.objects.get(pk=export_file_id)
     export_gift_cards(export_file, scope, file_type, delimiter)
+
+
+@app.task
+def delete_old_export_files():
+    now = timezone.now()
+
+    events = ExportEvent.objects.filter(
+        type=ExportEvents.EXPORT_PENDING,
+        date__lte=now - settings.EXPORT_FILES_TIMEDELTA,
+    ).values("export_file_id")
+    export_files = ExportFile.objects.filter(
+        Q(events__isnull=True) | Exists(events.filter(export_file_id=OuterRef("id")))
+    )
+
+    if not export_files:
+        return
+
+    paths_to_delete = list(export_files.values_list("content_file", flat=True))
+
+    for path in paths_to_delete:
+        if path:
+            default_storage.delete(path)
```

Comment:
I'm wondering if we should secure it a bit in case it raises `FileNotFoundError` / `OSError`? might there be a case when the path doesn't actually exists?

Reply:
I checked with S3 and default django storage and it's not raising an error when path does not exist. But I will consider adding it as I will be updating the counter.

Reply:
And IMO there is no reason for raising an error in such cases.